File: Jim/src/sk/fiit/jim/agent/highskill/Beam.java
package sk.fiit.jim.agent.highskill;

import sk.fiit.jim.agent.communication.Communication;
import sk.fiit.jim.agent.models.AgentModel;
import sk.fiit.jim.agent.moves.LowSkill;
import sk.fiit.jim.agent.skills.HighSkill;
import sk.fiit.robocup.library.geometry.Vector3D;

/**
 *
 * Reimplemented from Ruby to Java by roman moravcik
 */

public class Beam extends HighSkill {

    private Vector3D position;
    private boolean isRunning;
    private AgentModel agentModel = AgentModel.getInstance();  

    public Beam(Vector3D position) {
    	super();
    	this.isRunning = false;
    	this.position = position;
    	agentModel.setPosition(position);
    }

    @Override
    public LowSkill pickLowSkill() {    	
        /*
    	if (!isRunning) {
        	//LowSkill currentSkill = LowSkills.get("rollback");
        	this.isRunning = true;
            return currentSkill;
        } 
*/
        double yPos = position.getY();
        double xPos = position.getX();

        Communication.getInstance().addToMessage("(beam " + Double.toString(xPos) + " " + Double.toString(yPos) + " 0.0)");
        //     ypos = 1.0 + Java::sk.fiit.jim.agent.AgentInfo.playerId;
        //     Java::sk.fiit.jim.agent.communication.Communication.instance.addToMessage "(beam -5.0 " + ypos.to_s + " 0.0)"
        return null;
    }

    @Override
    public void checkProgress() {
    }
}

File: Old_xml_and_scripts/Jim/high_skills/old-predUcesanim/beam.rb
require 'high_skills/ruby_high_skill.rb'

class Beam < RubyHighSkill
  
  def initialize position
    super()
    @run_already = false
    @formationPosition = position 
#    @position = Java::sk.fiit.jim.agent.models.TacticalInfo.getInstance.getFormPositionGlobal()
  end
  
  def pickLowSkill
     unless @run_already then
          current_skill = get_skill "rollback"
          @run_already = true
          return current_skill
     end
     ypos = @formationPosition.getY()
     xpos = @formationPosition.getX()
     Java::sk.fiit.jim.agent.communication.Communication.instance.addToMessage "(beam " + xpos.to_s + " " + ypos.to_s + " 0.0)"
#     ypos = 1.0 + Java::sk.fiit.jim.agent.AgentInfo.playerId;
#     Java::sk.fiit.jim.agent.communication.Communication.instance.addToMessage "(beam -5.0 " + ypos.to_s + " 0.0)"
     nil
  end
 
  def checkProgress
    #do nothing
  end
end